app/lib/actions/candlepin/environment/set_content.rb

```diff
@@ -52,7 +52,12 @@ def finalize # rubocop:disable Metrics/AbcSize
         def existing_ids
           ::Katello::Resources::Candlepin::Environment.
               find(input[:cp_environment_id])[:environmentContent].map do |content|
-            content[:content][:id]
+            # Supports Candlepin 4.2.13 and older versions
+            if content.has_key?('contentId')
```

Comment:
I wonder what's faster: this or using `content.fetch()`. Perhaps also move the comment into the `else` branch so you know which branch is what.

Reply:
I like the readability here compared to `content.fetch('contentId', content[:content][:id])` and I'm guessing the difference in timing will be negligible. Plus with some added comments I think it'll be quite nice.